apps/meteor/ee/app/livechat-enterprise/server/hooks/onSaveVisitorInfo.ts

```diff
@@ -0,0 +1,65 @@
+import type { IOmnichannelRoom, IUser } from '@rocket.chat/core-typings';
+import { OmnichannelServiceLevelAgreements } from '@rocket.chat/models';
+
+import { callbacks } from '../../../../../lib/callbacks';
+import { removePriorityFromRoom, updateRoomPriority } from '../api/lib/priorities';
+import { removeRoomSLA, updateRoomSLA } from '../api/lib/sla';
+import { cbLogger } from '../lib/logger';
+
+const updateSLA = async (room: IOmnichannelRoom, user: IUser, slaId?: string) => {
+	if (!slaId) {
+		await removeRoomSLA(room._id, user);
+		return;
+	}
+
+	const sla = await OmnichannelServiceLevelAgreements.findOneById(slaId);
```

Comment:
if you just want to check if a record exists, pls use a `projection` to fetch only the `_id` field.